js/modules/k6/experimental/fs/file.go

```diff
@@ -55,7 +58,7 @@ func (f *file) Read(into []byte) (n int, err error) {
 
 	n = copy(into, f.data[start:end])
 
-	f.offset += n
+	f.offset.Store(offset + int64(n))
```

Comment:
```suggestion
	f.offset.Store(end)
```
or do we expect to be different?

Reply:
I have revamped the function completely to make it more readable and cater to your comment 🙇🏻